helpers/ruby

```diff
@@ -0,0 +1,310 @@
+export RBENV_ROOT="$rbenv_install_dir"
+export rbenv_root="$rbenv_install_dir"
+
+ruby_dependencies=""
+build_ruby_dependencies="libjemalloc-dev curl build-essential libreadline-dev zlib1g-dev libsqlite3-dev libssl-dev libxml2-dev libxslt-dev autoconf automake bison libtool"
+pkg_dependencies="$pkg_dependencies $ruby_dependencies"
+build_pkg_dependencies="$build_pkg_dependencies $build_ruby_dependencies"
+
+# Load the version of Ruby for an app, and set variables.
+#
+# ynh_use_ruby has to be used in any app scripts before using Ruby for the first time.
+# This helper will provide alias and variables to use in your scripts.
+#
+# To use gem or Ruby, use the alias `ynh_gem` and `ynh_ruby`
+# Those alias will use the correct version installed for the app
+# For example: use `ynh_gem install` instead of `gem install`
+#
+# With `sudo` or `ynh_exec_as`, use instead the fallback variables `$ynh_gem` and `$ynh_ruby`
+# And propagate $PATH to sudo with $ynh_ruby_load_path
+# Exemple: `ynh_exec_as $app $ynh_ruby_load_path $ynh_gem install`
+#
+# $PATH contains the path of the requested version of Ruby.
+# However, $PATH is duplicated into $ruby_path to outlast any manipulation of $PATH
+# You can use the variable `$ynh_ruby_load_path` to quickly load your Ruby version
+#  in $PATH for an usage into a separate script.
+# Exemple: $ynh_ruby_load_path $final_path/script_that_use_gem.sh`
+#
+#
+# Finally, to start a Ruby service with the correct version, 2 solutions
+#  Either the app is dependent of Ruby or gem, but does not called it directly.
+#  In such situation, you need to load PATH
+#    `Environment="__YNH_RUBY_LOAD_PATH__"`
+#    `ExecStart=__FINALPATH__/my_app`
+#     You will replace __YNH_RUBY_LOAD_PATH__ with $ynh_ruby_load_path
+#
+#  Or Ruby start the app directly, then you don't need to load the PATH variable
+#    `ExecStart=__YNH_RUBY__ my_app run`
+#     You will replace __YNH_RUBY__ with $ynh_ruby
+#
+#
+# one other variable is also available
+#   - $ruby_path: The absolute path to Ruby binaries for the chosen version.
+#
+# usage: ynh_use_ruby
+#
+# Requires YunoHost version 3.2.2 or higher.
+ynh_use_ruby () {
+    ruby_version=$(ynh_app_setting_get --app=$app --key=ruby_version)
+
+    # Get the absolute path of this version of Ruby
+    ruby_path="$ruby_version_path/$YNH_APP_INSTANCE_NAME/bin"
+
+    # Allow alias to be used into bash script
+    shopt -s expand_aliases
+
+    # Create an alias for the specific version of Ruby and a variable as fallback
+    ynh_ruby="$ruby_path/ruby"
+    alias ynh_ruby="$ynh_ruby"
+    # And gem
+    ynh_gem="$ruby_path/gem"
+    alias ynh_gem="$ynh_gem"
+
+    # Load the path of this version of Ruby in $PATH
+    if [[ :$PATH: != *":$ruby_path"* ]]; then
+        PATH="$ruby_path:$PATH"
+    fi
+    # Create an alias to easily load the PATH
+    ynh_ruby_load_path="PATH=$PATH"
+
+    # Sets the local application-specific Ruby version
+    pushd $final_path
```

Comment:
``
```suggestion
    pushd $install_dir
```

Reply:
(But we should probably have a trick to make the helper both packaging v1 and v2 compatible)

Reply:
```suggestion
    pushd ${install_dir:-$final_path}
```